drbert/data/dataset_readers.py

```diff
@@ -0,0 +1,242 @@
+from torchtext import data
+
+
+class DatasetReader(object):
+    """The parent class for all dataset readers.
+
+    Args:
+        path (str): Common prefix of the splits’ file paths.
+        partitions (dict): A dictionary keyed by partition ('train', 'test', 'validation')
+            containing the suffix to add to `path` for that partition.
+        tokenizer (transformers.PreTrainedTokenizer): The function used to tokenize strings into
+            sequential examples.
+        format (str): The format of the data file. One of “CSV”, “TSV”, or “JSON” (case-insensitive).
+        batch_sizes (tuple): Optional, tuple of batch sizes to use for the different splits, or None
+            to use the same batch_size for all splits. Defaults to None.
+        maxlen (int): Optional, a fixed length that all examples using this field will be padded to,
+            for flexible sequence lengths. Default: 512.
+        lower_case: (bool): Optional, True if text should be lower cased, False if not. Defaults to
+            False.
+
+    Raises:
+        ValueError: If batch_sizes is not a tuple or None.
+        ValueError: If batch_sizes is not None and `len(batch_sizes) != len(partitions)`.
+    """
+    def __init__(self, path, partitions, tokenizer, format, batch_sizes=None, maxlen=512,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4302662)